### PR TITLE
Revise how cache is computed

### PR DESCRIPTION
**Pull request by taalexander**
Fixes current CI build time issues resulting from cache misses.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -71,15 +71,17 @@ jobs:
...
+              all_in_cache=true
           else
-              cache_hit=true
+              all_in_cache=false
           fi
-          echo "cache_hit=$cache_hit" >> $GITHUB_OUTPUT
-          echo "Cache was hit: $cache_hit"
+          echo "all_in_cache=$all_in_cache" >> $GITHUB_OUTPUT
+          echo "Conan build requires:"
+          echo $build_requires
```
Comment (blakejohnson) [inline .github/workflows/lint.yml:83]: To be consistent with the next line, you might consider
```suggestion
          echo "Conan build requires: $build_requires"
```

**Review by blakejohnson — DISMISSED** [review]

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -71,15 +71,17 @@ jobs:
         working-directory: build
         run: |
           export CONAN_LLVM_GIT_CACHE="${{ runner.temp }}/llvm-project"
-          conan install .. -pr:h default -pr:b default
-          if [ $? -ne 0 ]
+          conan info .. -pr:h default -pr:b default --build=missing
+          build_requires=$(conan info .. -pr:h default -pr:b default --build=missing)
```
Comment (blakejohnson) [inline .github/workflows/lint.yml:75]: Why are you running the same `conan info` command twice?

Comment (taalexander) [reply .github/workflows/lint.yml]: Print debugging and comparing the stdout/stderr. This specific instance is a typo.

**Review by blakejohnson — APPROVED** [review]
Nice

**Pull request merged**